app/workers/notifications/schedule_reminder_mails_job.rb

```diff
@@ -0,0 +1,46 @@
+#-- encoding: UTF-8
+
+#-- copyright
+# OpenProject is an open source project management software.
+# Copyright (C) 2012-2021 the OpenProject GmbH
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License version 3.
+#
+# OpenProject is a fork of ChiliProject, which is a fork of Redmine. The copyright follows:
+# Copyright (C) 2006-2013 Jean-Philippe Lang
+# Copyright (C) 2010-2013 the ChiliProject Team
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+#
+# See docs/COPYRIGHT.rdoc for more details.
+#++
+
+module Notifications
+  class ScheduleReminderMailsJob < Cron::CronJob
+    # runs every hour, so 00:00, 01:00...
+    self.cron_expression = '0 * * * *'
```

Comment:
But that would result in an email being sent in the following very likely scenario:

* Notification being added at 7:50
* Job runs at 8:00 including the notification
* Notification being added at 8:10
* Job runs at 8:15 including the notification

This would result in two reminder mails being sent out within the same hour.

We could prevent this by limiting the latest notification to be included to have to be created before the hour the job runs for (e.g. before 8:00) but I don't know if that is worth it.

Reply:
I think I can add something to the call to `unsent_reminders_before` to only include notifications that where added before the full hour to prevent the scenario described above and still support those quarter hour time zones.

Reply:
I see your point. Not worth the huzzle, I believe.

Reply:
> I think I can add something to the call to `unsent_reminders_before` to only include notifications that where added before the full hour to prevent the scenario described above and still support those quarter hour time zones.

It's your call.

Reply:
Actually you were right, @wielinde. The code before did not work for e.g. Kathmandu with its +5:45 offset. I hope to have fixed this now. The cron job runs each 15 min now. If I am not mistaken, with the changes now in, this should not lead to the scenario I described above since we no longer floor to the previous full hour.